sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/red_team_agent/red_team_agent.py

```diff
@@ -1090,122 +1239,126 @@ async def attack(
-                            all_prompts=prompts_to_use,
+                            all_prompts=attack_objectives,
                             progress_bar=progress_bar,
                             progress_bar_lock=progress_bar_lock,
                             evaluation_name=evaluation_name,
                             data_only=data_only,
                             output_path=output_path,
                             risk_category=risk_category
                         )
-                    )
-                
-                # Process results for this risk category
-                self.logger.info(f"Gathering {len(risk_tasks)} tasks for risk category {risk_category.value}")
-                risk_results = await asyncio.gather(*risk_tasks)
-                self.logger.info(f"Received {len(risk_results)} results for risk category {risk_category.value}")
-                all_results.extend(risk_results)
+                        self.logger.info(f"Completed attack for {risk_cat_value} with strategy {strategy_name}")
+                        all_results.append(attack_result)
+                    except Exception as e:
+                        self.logger.error(f"Error running attack for {risk_cat_value} with strategy {strategy_name}: {str(e)}")
+                        # Update progress bar even if there's an error
+                        async with progress_bar_lock:
+                            progress_bar.update(1)
                 
-                # TODO: Calculate per-risk category metrics here if needed
-                self.logger.info(f"Completed processing for risk category: {risk_category.value}")
+                self.logger.info(f"Completed processing for risk category: {risk_cat_value}")
                 
             progress_bar.close()
             
             # Merge all results across risk categories
             self.logger.info("=" * 80)
             self.logger.info(f"PROCESSING RESULTS")
-            self.logger.info(f"Merging {len(all_results)} results across all risk categories")
+            self.logger.info(f"Merging {len(all_results)} results across all risk categories and strategies")
             merged_results = {}
             for d in all_results:
                 merged_results.update(d)
             self.logger.info(f"Merged results contain {len(merged_results)} entries")
-
             if data_only: 
                 self.logger.info("Data-only mode, returning merged results without evaluation")
                 return merged_results
             
-            # TODO: Implement aggregation of metrics across risk categories
-            # TODO: Consider how to properly weight and combine metrics from different risk categories
-            
+            # Convert results to RedTeamAgentResult format
             self.logger.info("Converting results to RedTeamAgentResult format")
             red_team_agent_result = self._to_red_team_agent_result(cast(Dict[str, EvaluationResult], merged_results))
```

Comment:
Please see latest changes in my branch for new output format and make sure this aligns - don't think there will be any issues though.